### ISSUE TITLE
Shifting to a Structured Header-like syntax?

### ISSUE THREAD

**Issue by mikewest** [WICG/signature-based-sri#18]
If we adopt the suggestion in #16 to shift towards RFC9421's `Identity-Digest`/`Signature-Input`/`Signature` headers, we'll be representing the ~same information we need in HTML, but using a different syntax.

I wonder whether it would be helpful for developers to adopt the same syntax in HTML as well (though of course we'd need to support the current syntax ~forever). That is, rather than:

```html
<script src="https://my.cdn/script.js"
        crossorigin="anonymous"
        integrity="sha256-[base64-encoded-hash]
                   sha512-[base64-encoded-hash]
                   ed25519-[base64-encoded-public-key]"></script>
```

we could recommend:

```html
<script src="https://my.cdn/script.js"
        crossorigin="anonymous"
        integrity="sha-256=:[base64-encoded-hash]:,
                   sha-512=:[base64-encoded-hash]:,
                   ed25519=:[base64-encoded-public-key]:"></script>
```

This seems worth considering from an aesthetic and consistency perspective, but I'm not sure the churn is actually worth it. 🤷

**Comment by LPardue** [comment]
Is the proposal for "like" Structured Fields or actually structured fields? Mainly thinking from a generator/parser perspective. Having something similar to Structured Fields but slightly different adds another potential source of interop failure. For example, would you be defining the `integrity` field as a [Dictionary](https://httpwg.org/specs/rfc9651.html#dictionary), where items are allowed to have additional [parameters](https://httpwg.org/specs/rfc9651.html#item)? 

I'm not sure the consistency of _presentation_ would actually help that much. So long as mapping between the two is well defined (which seems simple to do).

**Comment by LPardue** [comment]
Also, it's potentially confusing since the digest field Dictionary entries are strictly hashing <algorithm name>=<base 64 calculated hash>. The ed25519 entry in your example is neither of those things right?

Could probably be wordmithed to avoid confusion but  not sure.

**Comment by mikewest** [comment]
Right. The Ed25519 entry is a base64-encoded public key, not a content hash. But I think we'd also represent it as a byte sequence, and it doesn't seem terribly confusing to me to do so next to hashes (assuming that folks recognize that hashing algorithms and signature algorithms are distinct (which I think they have to in order to engage with this proposal no matter how we spell it)).

**Comment by LPardue** [comment]
I think the important distinction to make would be that the `integrity` property is **not** a structured field dictionary,  and does not represent what a single header field would contain. 

Instead `integrity` property is a comma-separated list of key value pairs where the value is a structured fields Byte Sequence Item. That would be fine IMO. We'd probably want some text to say that the key values in the property are related to the values registered by IANA.

**Comment by mikewest** [comment]
I'm happy to frame it in any way that makes sense.

That said, I'm still not sure it _does_ make sense. :) Let's work everything else out first.

**Comment by LPardue** [comment]
SGTM I could live with either format.

**Comment by mikewest** [comment]
We've made this shift. :)

**Comment by mikewest** [comment]
After a few months: it seems that there's little advantage in changing the attribute's syntax, and consistency with the headers doesn't seem actually important to support. I'm going to close this out, as I don't think we're going to do this.
